src/__Private/Visitor/VariableInfo.hack

```diff
@@ -0,0 +1,53 @@
+namespace Slack\GraphQL\__Private;
+
+use namespace Graphpinator\Parser;
+
+final class VariableInfo extends ASTVisitor {
+    private ?Parser\Operation\Operation $current_operation = null;
+    private ?Parser\Fragment\Fragment $current_fragment = null;
+
+    private dict<int, vec<Parser\Value\VariableRef>> $operation_to_variables = dict[];
+    private dict<int, vec<Parser\Value\VariableRef>> $fragment_to_variables = dict[];
+
+    public function __construct(private FragmentInfo $fragment_info) {}
```

Comment:
Don't love that we need to inject FragmentInfo here. Maybe there's a better way

Reply:
You're right, these do seem pretty similar (e.g. both need to track current operation/fragment), merging them into DependencyInfo like you suggested seems reasonable.